File: scsc/contrib/vax-scs/scs.c

/*  Classifier system:
	as found in Appendix D of "Genetic Algorithms in Search, Optimization,
	and Machine Learning" by David E. Goldberg.  Rewritten in C language by
	Erik Mayer   emayer@uoft02.utoledo.edu
*/


/* include files */
#include "declare.scs"
#include "io.scs"
#include "utility.scs"
#include "environ.scs"
#include "detector.scs"
#include "perform.scs"
#include "aoc.scs"
#include "effector.scs"
#include "reinforc.scs"
#include "timekeep.scs"
#include "advance.scs"
#include "ga.scs"
#include "report.scs"
#include "initial.scs"

main ()
{
    int iter;

    printf ("Enter random number seed (0-728): ");
    scanf ("%d", &seed);
    initialization ();
    detectors ();
    report ();
    for (iter = 0; iter <= timekeeprec.iterationsbrigade; ++iter) {
	timekeeper ();
	environment ();
	detectors ();
	matchclassifiers (envmessage);
	aoc ();
	effector ();
	reinforcement ();
	if (timekeeprec.reportflag)
	    report ();
	if (timekeeprec.consolereportflag)
	    consolereport ();
	if (timekeeprec.plotreportflag)
	    plotreport ();
	advance ();
	if (timekeeprec.gaflag) {
	    ga ();
	    if (timekeeprec.reportflag)
		reportga ();
	}
    }
    report ();
    fclose (pfile);
}

File: simpleCS/Pascal/SCS/scs.pas
program scs;

{ SCS -  A Simple Classifier System }
{   (C)  David E. Goldberg, 1987    }
{        All Rights Reserved        }

{$I declare.scs }
{$I random.apb }
{$I io.scs }
{$I utility.scs }
{$I environ.scs }
{$I detector.scs }
{$I perform.scs }
{$I aoc.scs }
{$I effector.scs }
{$I reinforc.scs }
{$I timekeep.scs }
{$I advance.scs }
{$I ga.scs }
{$I report.scs }
{$I initial.scs }

begin { main }
  initialization;
  detectors(environrec, detectrec, envmessage);
  report(rep);
  with timekeeprec do repeat
    timekeeper(timekeeprec);
    environment(environrec);
    detectors(environrec, detectrec, envmessage);
    matchclassifiers(population, envmessage, matchlist);
    aoc(population, matchlist, clearingrec);
    effector(population, clearingrec, environrec);
    reinforcement(reinforcementrec, population, clearingrec, environrec);
    if reportflag then report(rep);
    if consolereportflag then  consolereport(reinforcementrec);
    if plotreportflag then plotreport(pfile, reinforcementrec);
    advance(clearingrec);
    if gaflag then begin
      ga(garec, population);
      if reportflag then reportga(rep, garec, population);
     end;
  until halt;
  report(rep);  { final report }
  close(pfile); { close plot file }
end.

